### PR TITLE
Add API Security endpoint discovery tests

### PR DESCRIPTION
**Pull request by jandro996**
## Motivation

https://docs.google.com/document/d/1txwuurIiSUWjYX7Xa0let7e49XKW2uhm1djgqjl_gL0/edit?tab=t.0 

## Changes

Test that all payloads match the defined schema
Test the mandatory fields for each endpoint
One test per optional endpoint field
Test is_first flag

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed and the CI green, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/appsec/api_security/test_endpoint_discovery.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+@features.api_security_endpoint_discovery
+class Test_Endpoint_Discovery:
+    _main_setup_done: bool = False
+
+    def main_setup(self):
+        """Setup (once) for endpoint discovery tests."""
+        if Test_Endpoint_Discovery._main_setup_done:
+            return
+        for _ in range(10):
+            weblog.get("/")
```
Comment (IlyasShabi) [inline tests/appsec/api_security/test_endpoint_discovery.py:17]: why we need to do some calls?

File: tests/appsec/api_security/test_endpoint_discovery.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+            endpoints.extend(payload.get("endpoints", []))
+
+        assert endpoints, "No endpoints discovered"
+        return endpoints
+
+    def setup_endpoint_discovery(self):
+        """Setup for endpoint discovery test."""
+        self.main_setup()
+
+    def test_endpoint_discovery(self):
```
Comment (IlyasShabi) [inline tests/appsec/api_security/test_endpoint_discovery.py:49]: I think it's better to check endpoints returned by telemetry to test some edge cases like GET/HEAD and *

Comment (christophe-papazian) [reply tests/appsec/api_security/test_endpoint_discovery.py:49]: I'm not sure to understand what you mean. We are already checking endpoints returned by telemetry, but as the tests are defined for all weblogs, we can't make assumptions on which method are allowed here.
Maybe we could add an endpoint to the weblog specification that only accept GET to check for this particular case? (This could cause problem to frameworks that accept all methods and can't restrict them, if there are any)

Comment (IlyasShabi) [reply tests/appsec/api_security/test_endpoint_discovery.py:49]: IMO we should test some edge cases like wildcard routes `*` mentioned in the RFC for endpoints that accept all methods if it's available for all languages.

> Maybe we could add an endpoint to the weblog specification that only accept GET to check for this particular case?

Yes we can do this also

**Review by IlyasShabi — APPROVED** [review]
